### ISSUE TITLE
Missing PyPy Wheels

### ISSUE THREAD

**Issue by jcpunk** [sqlalchemy/sqlalchemy#6702]
**Describe the bug**
The publication of SQLAlchemy-1.4.20 didn't result in any PyPy compatible wheels.  Wheels for CPython were generated.  Since setup.cfg reports PyPy support, I'd expected to see wheels up there for PyPy.

**To Reproduce**
https://pypi.org/project/SQLAlchemy/#files

https://github.com/sqlalchemy/sqlalchemy/blob/master/setup.cfg#L29

**Error**
```
$ pypy3 -m pip install --user sqlalchemy
Collecting sqlalchemy
  Using cached SQLAlchemy-1.4.20.tar.gz (7.7 MB)
Requirement already satisfied: importlib-metadata in ./.local/lib/pypy3.7/site-packages (from sqlalchemy) (4.6.0)
Requirement already satisfied: greenlet!=0.4.17 in /usr/lib64/pypy3-7.3/lib_pypy (from sqlalchemy) (0.4.13)
Requirement already satisfied: zipp>=0.5 in ./.local/lib/pypy3.7/site-packages (from importlib-metadata->sqlalchemy) (3.4.1)
Requirement already satisfied: typing-extensions>=3.6.4 in ./.local/lib/pypy3.7/site-packages (from importlib-metadata->sqlalchemy) (3.10.0.0)
Building wheels for collected packages: sqlalchemy
  Building wheel for sqlalchemy (setup.py) ... done
  Created wheel for sqlalchemy: filename=SQLAlchemy-1.4.20-pp37-pypy37_pp73-linux_x86_64.whl size=1484454 sha256=75ec017cd3b82682ea53346494c749c0a699a34f2a6125f9578ae905e36047ec
  Stored in directory: /home/jcpunk/.cache/pip/wheels/b9/45/0c/ac4d449cac940a2e2ef4e919111cbbb88ba6309c15a6403f5f
Successfully built sqlalchemy
Installing collected packages: sqlalchemy
Successfully installed sqlalchemy-1.4.20
```

**Versions.**
 - OS: Fedora 34
 - Python: PyPy 7.3.4 with GCC 11.1.1 20210512 (Red Hat 11.1.1-2) on linux
 - SQLAlchemy: 1.4.20
 - Database: SQLLite
 - DBAPI: SQLLite

**Additional context**

**Have a nice day!**

**Comment by CaselIT** [comment]
Hi,

I don't think we have ever published wheels without compiled extension, so I would guess that's always been the case.
Can you confirm that the same issue happens when installing an older version, like 1.4.10?

I guess we could build a wheel with no c-extension so that it takes the tag any-none, if it's not preferred by pip over the other for platform that support it. A drawback of this would be that platform we don't target would not get the a change to build the compiled extension, since they would by default install from the any-none wheel and not from source

**Comment by jcpunk** [comment]
I took a look over the release history of the `1.4` branch and am not seeing any wheels for PyPy.

**Comment by CaselIT** [comment]
Do the pypy wheel have a particular tag or they just use the any-none one?

**Comment by jcpunk** [comment]
For pypy specific bits, pp37-pypy37_pp73 for the latest pypy binary.  For noarch wheels, `any-none` should be sufficient.

To my knowledge all the Database C extensions (I've tried) work against PyPy.

**Comment by CaselIT** [comment]
> To my knowledge all the Database C extensions (I've tried) work against PyPy.  I'd be happy what whatever `pypy3 setup.py bdist_wheel` produces.

I would guess that it would be counterproductive to compile c-extension on pypy. I'll look into what the default tag is when creating them.

**Comment by CaselIT** [comment]
The wheel came out as `SQLAlchemy-1.4.26.dev0-pp37-pypy37_pp73-linux_x86_64.whl`, so I think we would still need to transform it to a manylinux one, even if no compiled stuff is in there, so it's effectively universal.

An alternative is to add an universal wheel with no c extension compiled that can be used a fallback, but we would need to change  this https://github.com/sqlalchemy/sqlalchemy/blob/2e6285d8d1d293c841f18313c94e493b2804055c/setup.py#L79-L87
I think that would be fine to do, since I think pip will prefer a platform specific wheel to the universal one (but I would need to check)

@zzzeek do you remember why that flag was forced to True?

**Comment by zzzeek** [comment]
iirc removing that flag meant the setup will always build a pure python wheel and not build C extensions.

**Comment by CaselIT** [comment]
I think we should be able to make it conditional on the flag that's used to tell to add the c files or not

**Comment by zzzeek** [comment]
is this resolved by the cbuildwheel thing?

**Comment by CaselIT** [comment]
Not 100% sure, but I think so

**Comment by CaselIT** [comment]
This is currently blocked by https://github.com/pypa/cibuildwheel/issues/1247 so we may not be able to do this for v2

also it seems that the test suite fails at least on windows. Will open a separate issue for it

**Comment by CaselIT** [comment]
Ok, since we do not build any extension on pypy a simple pure python wheel could be published to pypi for use in pypy.
Platform specific wheels are not need in this case.

**Comment by sqla-tester** [comment]
**Trevor Gross** has proposed a fix for this issue in the **main** branch:

**Use cibuildwheel to create wheels** https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4047
